### DISCUSSION TITLE
How do I change language of messages?

### DISCUSSION THREAD

**Discussion by AgainPsychoX** [microsoft/vscode-cpptools#9192] [Q&A]
I want to use English (which is used by entire VS Code except CppTools apparently) instead system wide Polish (on Windows 10).

![image](https://user-images.githubusercontent.com/23557839/163595913-ad70bf35-e3f2-4732-8ae6-9b12f6d30134.png)

I can't find easy switch. I don't think I have Polish language enabled anywhere for VS Code anyways too.

**Comment by bobbrow** [comment]
I believe you just need to run the `Configure Display Language` command and pick English (en)

**Comment by Colengms** [comment]
Hi @AgainPsychoX .  CppTools uses a mechanism provided by VS Code ( https://github.com/microsoft/vscode-nls, https://github.com/microsoft/vscode-nls-dev ) to select strings for the current language.  If the language you are seeing in CppTools strings differs from VS Code's, could you open a bug against VS Code for this?  (Which may end up being a bug in vscode-nls/vscode-nls-dev).

One scenario I've seen this happen in is when a user downloads a language pack extension, switches to that language, then removes the language pack.  VS Code reverts to English because the other language strings are no longer available, but does not change the language it provides to vscode-nls/vscode-nls-dev for CppTools.  That is tracked by: https://github.com/microsoft/vscode/issues/82791

It you haven't removed a language pack, perhaps there may be a similar VS Code bug where the default is simply non-English but VS Code displays its strings in English due to no language pack extension having been installed.  VS Code should be ensuring the language is consistent between itself and extensions using vscode-nls/vscode-nls-dev.

**Comment by MinusPL** [comment]
I can add to this, since it also happened to me.
VS Code here is huge mess. I installed VS Code and never installed any language pack. Same goes for installing extensions. And since I never had to change language I only assume it was defaulting to English pack, but internally was still using system default language (which for me is polish). After I installed language pack and changed language to polish and back to english it started properly showing english messages.

Workarounds for this issue are:
- set english as default locale in Windows.
- download any language pack, issue Change Display Language command to downloaded language and back to english.

I will also open an issue for that thing.

**Reply by MinusPL** [reply]
Forgot to ask: Since I am not sure if that is a bug related to vs code itself or this extension I'll wait for information where should I submit an issue.

**Reply by sean-mcmanus** [reply]
@MinusPL It would be a VS Code bug.

**Comment by wine-fall** [comment]
hi guys, I just follow someone advice, which is not reasonable but works
1. go to `Configure Display Language`
2. change the language to your native language
3. change the language back to English

**Comment by starball5** [comment]
Possibly related Stack Overflow question: [Why isn't my display language setting working everywhere with the VS Code Cpptools extension?](https://stackoverflow.com/q/75702473/11107541)

**Comment by YuchengWang** [comment]
I have the same problem. 

My VS Code's current language is Chinese. 
I get "en-us" from process.env.VSCODE_NLS_CONFIG.locale.

Output to  **console.log(process.env.VSCODE_NLS_CONFIG);**

`[Extension Host] {"locale":"en-us","availableLanguages":{"*":"zh-cn"},"_languagePackId":"a528b0de79e78fe9f26ab8283e474126.zh-cn","_translationsConfigFile":"/home/yorkwang/.config/Code/clp/a528b0de79e78fe9f26ab8283e474126.zh-cn/tcf.json","_cacheRoot":"/home/yorkwang/.config/Code/clp/a528b0de79e78fe9f26ab8283e474126.zh-cn","_corruptedFile":"/home/yorkwang/.config/Code/clp/a528b0de79e78fe9f26ab8283e474126.zh-cn/corrupted.info","_resolvedLanguagePackCoreLocation":"/home/yorkwang/.config/Code/clp/a528b0de79e78fe9f26ab8283e474126.zh-cn/92da9481c0904c6adfe372c12da3b7748d74bdcb","_languagePackSupport":true} (at console.<anonymous> (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:96:138784))`

**Comment by pai4451** [comment]
I also experienced same problem in 2025, and follow @wine-fall indeed works!

1. go to Configure Display Language
2. change the language to your native language
3. change the language back to English
